test/sharness/t0080-repo.sh

```diff
@@ -9,151 +9,191 @@ test_description="Test ipfs repo operations"
+		random 1000000 >multiblock &&
+		MBLOCKHASH=`ipfs add -q multiblock`
+	'
+
+	test_expect_success "'ipfs pin ls -type=indirect' is correct" '
+		ipfs refs "$MBLOCKHASH" >refsout &&
+		ipfs refs -r "$HASH_WELCOME_DOCS" >>refsout &&
+		sort refsout >refsout_sorted &&
+		ipfs pin ls -type=indirect >indirectpins &&
+		sort indirectpins >indirectpins_sorted &&
+		test_cmp refsout_sorted indirectpins_sorted
+	'
+
+	test_expect_success "pin something directly" '
+		echo "ipfs is so awesome" >awesome &&
+		DIRECTPIN=`ipfs add -q awesome` &&
+		echo "unpinned $DIRECTPIN" >expected9 &&
+		ipfs pin rm -r "$DIRECTPIN" >actual9 &&
+		test_cmp expected9 actual9 &&
+		echo "pinned $DIRECTPIN directly" >expected10 &&
+		ipfs pin add "$DIRECTPIN" >actual10 &&
+		test_cmp expected10 actual10
+	'
+
+	test_expect_success "'ipfs pin ls -type=direct' is correct" '
+		echo "$DIRECTPIN" >directpinexpected &&
+		DPHASH=QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn &&
+		echo "$DPHASH" >>directpinexpected &&
+		sort directpinexpected >dp_exp_sorted &&
+		ipfs pin ls -type=direct >directpinout &&
+		sort directpinout >dp_out_sorted &&
+		test_cmp dp_exp_sorted dp_out_sorted
+	'
+
+	test_expect_success "'ipfs pin ls -type=recursive' is correct" '
+		echo "$MBLOCKHASH" >rp_expected &&
+		echo "$HASH_WELCOME_DOCS" >>rp_expected &&
+		ipfs refs -r "$HASH_WELCOME_DOCS" >>rp_expected &&
+		sort rp_expected >rp_exp_sorted &&
+		ipfs pin ls -type=recursive >rp_actual &&
+		sort rp_actual >rp_act_sorted &&
+		test_cmp rp_exp_sorted rp_act_sorted
+	'
+
+	test_expect_success "'ipfs pin ls -type=all' is correct" '
+		cat directpinout >allpins &&
+		cat rp_actual >>allpins &&
+		cat indirectpins >>allpins &&
+		sort allpins >allpins_sorted &&
+		ipfs pin ls -type=all >actual_allpins &&
+		sort actual_allpins >actual_allpins_sorted &&
+		test_cmp allpins_sorted actual_allpins_sorted
+	'
+
+	test_expect_success "'ipfs pin rm' succeeds" '
+		ipfs pin rm "$DPHASH" &&
+		ipfs pin rm "$DIRECTPIN" &&
+		ipfs pin ls >rm_actual &&
+		test_cmp empty rm_actual &&
+		ipfs pin rm -r "$MBLOCKHASH" &&
+		ipfs pin rm -r "$HASH_WELCOME_DOCS" &&
+		ipfs pin ls -type=recursive >rm_actual &&
+		test_cmp empty rm_actual
+	'
+
+	test_expect_success "'ipfs repo gc' succeeds" '
+		ipfs repo gc >gc_out_actual
+	'
+
+	test_expect_success "'ipfs repo gc' looks good" '
+		echo "removed $DIRECTPIN" >gc_out_expected &&
+		echo "removed $DPHASH" >>gc_out_expected &&
+		echo "removed $HASH_WELCOME_DOCS" >>gc_out_expected &&
+		echo "removed $MBLOCKHASH" >>gc_out_expected &&
+		sort gc_out_actual >gc_out_actual_sorted &&
+		sort gc_out_expected >gc_out_expected_sorted &&
+		test_cmp gc_out_expected_sorted gc_out_actual_sorted
+	'
+
+}
+
+# should work offline
+test_repo_cmd
+
+# should work online...
+# ...but unfortunately doesn't work right now
+# there are some "Error: context deadline exceeded"
+# when running the online tests after the offline tests
+# if the offline tests are commented out, it works
```

Comment:
[This](http://gateway.ipfs.io/ipfs/QmdRLMUTSFfFkYnf4kBsQwqTWxBTSFWLsDT9iVadcm2GrH) is the `trashDirectory.t9999-onlinedebug.sh` directory after a run of the above test with debug nocolor logging enabled. The shell output is [here](http://gateway.ipfs.io/ipfs/QmR2yuX84jjVb5UorBCW2MGPWAwMQ32z9TSDnvJTiJBy4T)

It looks like the daemon somehow hangs before opening up the HTTP endpoints.